firmware/smbios/smbios.ksy

```diff
@@ -0,0 +1,37 @@
+meta:
+  id: smbios
+  endian: le
+  imports:
+    - smbios_bios_info
+    - smbios_system_info
+    - smbios_processor_info
+    - smbios_portable_battery
+    - smbios_skipper
+    - smbios_strings
+seq:
+  - id: tables
+    type: table
+    repeat: eos
+types:
+  table:
+    seq:
+      - id: type
+        type: u1
+        enum: type_enum
+      - id: table
+        type:
+          switch-on: type
+          cases:
+            # TODO: Add rest of the tables
+            'type_enum::table0': smbios_bios_info
+            'type_enum::table1': smbios_system_info
+            'type_enum::table4': smbios_processor_info
+            'type_enum::table22': smbios_portable_battery
+            _: smbios_skipper
```

Comment:
Ideally, "skipper" should go away, being just the byte array in this case, if all the other common parts of all tables would be extracted to this level.